retry.go

```diff
@@ -0,0 +1,72 @@
+package squalor
+
+import (
+	"fmt"
+	"github.com/go-mysql/conn"
+	"github.com/go-sql-driver/mysql"
+	"net"
+	"time"
+)
+
+var sleeper = time.Sleep
+
+type Retryable interface {
+	Retry(fn func() error) error
+}
+
+type NoOpConfiguration struct{}
+
+func (n *NoOpConfiguration) Retry(fn func() error) error {
+	return fn()
+}
+
+type RetryConfiguration struct {
+	Retries                     int
+	SleepBetweenRetriesInMillis int64
+	RetryableExceptions         map[error]bool
+}
+
+var BasicRetryConfiguration = RetryConfiguration{
+	3,
+	300,
+	map[error]bool{conn.ErrReadOnly: true, conn.ErrConnLost: true, conn.ErrConnCannotConnect: true, conn.ErrTimeout: true, mysql.ErrInvalidConn: true}}
+
+func (retryConfig *RetryConfiguration) Retry(fn func() error) error {
+	var err error
+	for i := 1; ; i++ {
```

Comment:
Can you just include `i < retryConfig.Retries` as part of the loop control?

Reply:
Done.